sale_stock_cancel_restriction/tests/test_sale_stock_cancel_restriction.py

```diff
@@ -25,5 +24,45 @@ def setUpClass(cls):
         cls.picking.button_validate()
 
     def test_cancel_sale_order(self):
-        with self.assertRaises(UserError):
-            self.sale_order.action_cancel()
+        """
+        Test a SO with a product on delivery with a 5 quantity.
+        Create two invoices: one for 3 quantity and one for 2 quantity
+        Then cancel Sale order, it won't raise any warning, it should be cancelled.
+        """
+        partner = self.partner
```

Comment:
Why are you creating another sales order if the setUp already contains one and its picking? Previous test code was correct.